### ISSUE TITLE
Tensor Error (Type and Dimension)

### ISSUE THREAD

**Issue by lucky7323** [lucky7323/DL4every1_PyTorch#1]
**발생시점**
Lecture 9의 kaggle-otto dataset을 이용한 softmax classifier 예제 코드 작성 中

**상황**
loadtxt로 csv파일을 옵션을 이용해 float data type의 ndarray로 바꾸어 dataset을 만들었다.
`xy = np.loadtxt(data, delimiter=',', skiprows=1, dtype=np.float32)`

그렇게 만든 ndarray를 torch.from_numpy를 이용해 다시 tensor로 바꾸었는데, 이 dataset을 사용하려니 loss function에서 LongTensor type을 받아야한다는 에러가 떴다. 또한 multi-dimension 사용이 불가능하다는 에러도 떴다.

**Comment by lucky7323** [comment]
**원인 및 해결방법**
1. CrossEntropyLoss function은 **input**으로 **FloatTensor**, **target**으로 **LongTensor** 를 받는다.
따라서 x는 그대로 float형으로 두고, y는 `y_data = y.type(torch.LongTensor)` 를 통해 LongTensor로 바꾸면 된다.

2. CrossEntropyLoss function은 one-hot vector가 아니라 index를 받는다. 이것까진 알고 있었고, 따라서
 batch size X 1의 class index를 가지는 행렬을 target으로 받는것이 문제가 없어보였다. 하지만 N X 1 행렬과 N 행렬은 다르다! (전자는 2차원이고, 후자는 1차원) 
따라서 N X 1 행렬을 N 행렬로 바꾸면 된다. pytorch에 구현된 **squeeze**란 함수를 사용해 쉽게 바꿀 수 있다. type문제와 동시에 해결하는 코드는 다음과 같다.
`self.y_data = torch.squeeze(y.type(torch.LongTensor))`
